### ISSUE TITLE
a 'clear" command would be nice

### ISSUE THREAD

**Issue by rdm** [jsoftware/j-playground#98]
Currently, we can clear the terminal session from the drop down menu, but not from within J.

**Comment by xelxebar** [comment]
This should already be possible. What happens if you hit ctrl-l in the session?

**Comment by rdm** [comment]
Nothing happens observable happens when I hit ctrl-I in the session.

But, also, hitting ctrl-I is not something I can do from within J.

**Comment by xelxebar** [comment]
@rdm Ctrl-L, not ctrl-I. This is a lower-case L: l. This in an upper-case i: I. I'm guessing those both look the same in your particular font.

> But, also, hitting ctrl-I is not something I can do from within J.

You have to help us help you. At the moment we have no idea how you're running J, what OS and environment you're running from, what version you're on, _etc_. Without any details, there's nothing we can do for you.

**Comment by rdm** [comment]
I'm currently using using Chrome Version 117.0.5938.134 (Official Build) (64-bit), on microsoft windows 11 home version 10.0.22621 Build 22621. But that's about to change.

That said, if a command were implemented in J, to remove all content from the J playground terminal window, why would that fail to work on some platforms?

Sure, hypothetically, I might be able to implement something using 2!:0 which manipulates the DOM underneath the J session. But that seems like it would be a modularity violation - it would introduce a dependence on the implementation which would likely interfere with future updates to the playground (and possibly breaking some existing playground mechanism, depending on what I tried to delete)).

That said control-L puts the focus on the url bar of chrome and selects the url such that if I hit the space bar the url would be replaced with a space character.

**Comment by xelxebar** [comment]
Ah, my bad. I didn't realize this was j-playground. Doesn't `)cls` do the trick? Or are you wanting a function that you can use programmatically?

**Comment by rdm** [comment]
You could try it yourself. (And I'm not sure what context you are reading this in, and replying in, where you're not aware that this is a j-playground issue.)

But, no, **)cls** gives me a syntax error in the j-playground.

**Comment by xelxebar** [comment]
`)cls` works in the URL linked in the project description: https://jsoftware.github.io/j-playground/bin/html/index.html.
However, that appears to be an old version.  Trying in the wasm port, I see the syntax error you're seeing: https://jsoftware.github.io/j-playground/bin/html2/#.
Are any of the documented shortcuts helpful to you? https://code.jsoftware.com/wiki/Playground/Shortcuts. The "Clear Term" one (Ctrl+Shift+T) should probably be displayed in the View menu, similar to other options.

**Comment by rdm** [comment]
Ah, that explains it. 

Ctrl+Shift+T brings up a new window in the instance of Chrome which I am using.
Ctrl+Shift+R works for me

But I do not know how to issue these keyboard shortcuts as a J command.

And my problem was not that _I_ did not know how to clear the terminal - the problem was that I did not know how to have _my J session_ clear the terminal.

(That said, I agree that updating the menu implementation to show these keyboard shortcuts would be a good idea.)

**Comment by xelxebar** [comment]
> the problem was that I did not know how to have my J session clear the terminal.

True. Looks like the shortcut just calls a `clearterm` function: https://github.com/jsoftware/j-playground/blob/6b06698a821472908ee9ad53209eee3c4a4a0eed/bin/html2/term.js#L17. Should be easy enough to expose that via some _bona fide_ prefix command _a la_ `)cls`.

More generally, maybe an escape hatch to call arbitrary JS is wanted? Could be a can of worms, though.

**Comment by rdm** [comment]
Ah, thanks.

And, yes, `2!:0'clearterm()'` does the trick.

Assuming that's a stable interface, this issue can be closed.

Thanks again.
